doc/source/whatsnew/v2.2.0.rst

```diff
@@ -473,6 +473,7 @@ Other Deprecations
 - Deprecated the ``kind`` keyword in :meth:`Series.resample` and :meth:`DataFrame.resample`, explicitly cast the object's ``index`` instead (:issue:`55895`)
 - Deprecated the ``ordinal`` keyword in :class:`PeriodIndex`, use :meth:`PeriodIndex.from_ordinals` instead (:issue:`55960`)
 - Deprecated the ``unit`` keyword in :class:`TimedeltaIndex` construction, use :func:`to_timedelta` instead (:issue:`55499`)
+- Deprecated the behavior of :meth:`DataFrame.replace` and :meth:`Series.replace` with :class:`CategoricalDtype`; in a future version replace will change the values while preserving the categories. To change the categories, use ``ser.cat.rename_categories`` instead (:issue:`55147`)
```

Comment:
Just curious. If one does `Categorical[int].replace(1, 1.2)` in the future will that case raise?

Reply:
It will behave like `ser[ser == 1] = 1.2`, which raises `TypeError: Cannot setitem on a Categorical with a new category (1.2), set the categories first`